python/pyspark/sql/dataframe.py

```diff
@@ -1980,6 +1981,11 @@ def dropDuplicates(self, subset=None):
         |Alice|  5|    80|
         +-----+---+------+
         """
+        if subset is not None and not isinstance(subset, Iterable):
```

Comment:
```suggestion
        if subset is not None and (
                not isinstance(subset, Iterable) or isinstance(subset, str)):
```

shall we combine the condition?

Reply:
sure, thought it was more legible if we separated, but combined looks good.